decimal.go

```diff
@@ -158,6 +158,18 @@ func NewFromFloatWithExponent(value float64, exp int32) Decimal {
 	}
 }
 
+// NewFromRat returns a decimal from a big.Rat representation.
+func NewFromRat(value *big.Rat) Decimal {
+	// fast path, where Rat is an int
+	if value.IsInt() {
+		f, _ := value.Float64()
```

Comment:
this only works for small Rats. big.Rats won't necessarily fit into a Float64, so you'll be losing information here.